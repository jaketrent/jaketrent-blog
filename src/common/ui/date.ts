export const currentYear = () => {
  return new Date().getFullYear()
}

export const formatHumanDate = (dateStr: string) => {
  const date = new Date(dateStr)
  const d = date.getDate()
  const mmm = date.toLocaleString("default", { month: "short" })
  const yyyy = date.getFullYear()

  return `${d} ${mmm} ${yyyy}`
}
