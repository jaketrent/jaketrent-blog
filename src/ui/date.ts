export function year(date = new Date()) {
  return date.getFullYear()
}
