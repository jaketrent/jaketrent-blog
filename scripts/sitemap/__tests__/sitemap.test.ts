import { generateSitemapIndex } from "../sitemap"
import { stabilizeDate } from "../../../test/util/date"

describe("generateSitemapIndex", () => {
  const teardownStabilizeDate = stabilizeDate(new Date("2018-01-01"))

  afterAll(() => {
    teardownStabilizeDate()
  })

  it("renders all static .tsx pages", async () => {
    const sitemap = await generateSitemapIndex()
    expect(sitemap).toMatchSnapshot()
  })
})
