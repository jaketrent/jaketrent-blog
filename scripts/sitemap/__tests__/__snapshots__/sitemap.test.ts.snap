// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSitemapIndex renders all static .tsx pages 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\"?>
<sitemapindex xmlns=\\"http://www.sitemaps.org/schemas/sitemap/0.9\\">
  <sitemap>
    <loc>https://jaketrent.com/sitemap-blog.xml.gz</loc>
    <lastmod>2018-01-01T00:00:00.000Z</lastmod>
  </sitemap>

  <sitemap>
    <loc>https://jaketrent.com/sitemap-static.xml.gz</loc>
    <lastmod>2018-01-01T00:00:00.000Z</lastmod>
  </sitemap>
</sitemapindex>
"
`;
